iac-extensions/arm/src/main/java/org/sonar/iac/arm/tree/impl/bicep/TypeDeclarationImpl.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.arm.tree.impl.bicep;
+
+import java.util.List;
+import org.sonar.iac.arm.tree.api.Identifier;
+import org.sonar.iac.arm.tree.api.StringLiteral;
+import org.sonar.iac.arm.tree.api.bicep.SyntaxToken;
+import org.sonar.iac.arm.tree.api.bicep.TypeDeclaration;
+import org.sonar.iac.arm.tree.impl.AbstractArmTreeImpl;
+import org.sonar.iac.common.api.tree.Tree;
+
+public class TypeDeclarationImpl extends AbstractArmTreeImpl implements TypeDeclaration {
+
+  private final SyntaxToken type;
+  private final Identifier name;
+  private final SyntaxToken equ;
+  private final StringLiteral typeExpression;
```

Comment:
General question about StringLiteral. Aren't we allowing to much?
In the grammar:
`literalValue -> NUMBER | "true" | "false" | "null"` doesn't allow any string.

Reply:
Yes, you are right.
My goal is not to make it correct, I'm using `StringLiteral` as a temporary placeholder until we implement `typeExpression`.
Allowing too much is also usually not an issue when writing our parsers, as we are expecting that the provided code is correct.